llvm/lib/CodeGen/TargetRegisterInfo.cpp

```diff
@@ -674,6 +674,50 @@ TargetRegisterInfo::prependOffsetExpression(const DIExpression *Expr,
                                       PrependFlags & DIExpression::EntryValue);
 }
 
+unsigned TargetRegisterInfo::getRegPressureSetLimit(const MachineFunction &MF,
```

Comment:
It's still desirable to cache this somewhere (which I guess was the point of RegisterClassInfo). Computing this is very much not cheap. Is there a way to try to push users to go through RegisterClassInfo?

Reply:
I don't think we have a way to avoid people from using `TargetRegisterInfo::getRegPressureSetLimit` directly as long as it is a public method...

Reply:
We can add a comment on `TargetRegisterInfo::getRegPressureSetLimit` to suggest people to use `RegisterClassInfo`, it may help in a way... And then, we can rewrite those direct users to use `RegisterClassInfo` as follow ups.